GRIDAPPSD/gridappsd-sample-distributed-app
# implemented neighboring agent discovery and agent to agent communication

This pull request:
1. Added a script to add a Context Manager agent at each topological layer.
2. Added to script to create a sample agent at each layer to demonstrate reqeust-response from a distributed service (for eg. local y bus).
3. Modified sample distributed app to discover neighboring agents and communicate with them. 

To test:
1. Run GridAPPS-D platform
2. Merge and use updated gridappsd-python from PR: https://github.com/GRIDAPPSD/gridappsd-python/pull/114
3. Execute run_context_mananger_agent.py
4. Execute run_sample_distributed_service.py 
5. Execute run_sample_distributed_app.py

Expected result:
Switch area agents in the sample app should 
- print a list of neighboring agents
- send a request and receive response from service agents
